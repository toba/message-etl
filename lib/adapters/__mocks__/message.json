{
   "participants": [
      {
         "name": "Other Person"
      },
      {
         "name": "Self Person"
      }
   ],
   "messages": [
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539713573381,
         "content": "Oh, I know what that means. Ok",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539713210175,
         "content": "Maybe I translated wrong. Full message: \"Its 3160 Lewis River Rd Woodland Wa 98674 for know it will be changing when the new houses and road goes in don't when that will happen.\"",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539713187186,
         "content": "What do you mean not in the new place?",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539713157344,
         "content": "Jessica: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539553590548,
         "content": "You sent a video.",
         "videos": [
            {
               "uri": "messages/OtherPerson_ea77ad6afb/videos/video1539553590_336614783757508.mp4",
               "creation_timestamp": 1539553586,
               "thumbnail": {
                  "uri": "messages/OtherPerson_ea77ad6afb/videos/thumbnails/42075454_2168518689894738_1310336861426507451_n.jpg"
               }
            }
         ],
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539712398807,
         "content": "And he spelled \"your\" wrong",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539465166225,
         "content": "You: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539712382725,
         "content": "Scott says the address is still 3160 Lewis River Rd. Not in the new place yet.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539705094711,
         "content": "Rough",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539704752058,
         "content": "Went to make latte for first time ever here and we're out of milk. What a dump.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539657694856,
         "content": "Jessica: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657676405,
         "content": "Was on perch a while",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657668688,
         "content": "Mellow",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657665741,
         "content": "Seems to be",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539657639664,
         "content": "She feeling okay?",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657593609,
         "content": "Cats new hiding spot",
         "reactions": [
            {
               "reaction": "\u00f0\u009f\u0098\u0086",
               "actor": "Other Person"
            }
         ],
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539657585171,
         "content": "You sent a photo.",
         "photos": [
            {
               "uri": "messages/JessicaAbbott_ea77ad6afb/photos/44126872_2446710928889289_825560690738593792_n_2446710902222625.jpg",
               "creation_timestamp": 1539657584
            }
         ],
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539653084097,
         "content": "So very sad",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539652968444,
         "content": "Felt so sad going in his room to turn off the light, looking around",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649902196,
         "content": "You: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539649893421,
         "content": "Oh, I know it was 3160. I'll find out",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649832938,
         "content": "Meant as more of a question :)",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539649769629,
         "content": "Thanks",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649670977,
         "content": "3170 Lewis River Rd\nWoodland, WA 98674",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539649669514,
         "content": "The address I have is",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539643699846,
         "content": "You: \u00f3\u00b0\u0080\u0080",
         "sticker": {
            "uri": "messages/stickers_used/39178562_1505197616293642_5411344281094848512_n_369239263222822.png"
         },
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539643632522,
         "content": "I'll be by to get Hunter around 5:30",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539643575006,
         "content": "Waiting in line for cat",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539633414962,
         "content": "yeah ...",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539633395761,
         "content": "no clue on the xbox, sold is probably a good bet. He can deal with Scott on that one",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539633266739,
         "content": "didn't see xbox anywhere. i guess he sold it? or I put it somewhere I totally forgot? but i don't think so. just had it in the office, locked.",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539633220098,
         "content": "checked his room. it seemed clean in the sense that there aren't clothes on the floor anymore.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1539633194653,
         "content": "If he's back by 4:30 and his room is clean and he's totally packed",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1539632575070,
         "content": "Hunter wanting to go visit friends. I said to check with you.",
         "type": "Generic"
      },
      {
         "sender_name": "Self Person",
         "timestamp_ms": 1371924211989,
         "content": "This is my route\nhttps://dl.dropboxusercontent.com/u/33199511/lava-mountain-loop.jpg",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1370924843625,
         "content": "Good night, sweet dreams reality is obviously not your forte.",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1370872998841,
         "content": "SPOILED monster.\n",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1370793302848,
         "content": "That is hilarious, we are all laughing. Hitting the road now",
         "type": "Generic"
      },
      {
         "sender_name": "Other Person",
         "timestamp_ms": 1298522821000,
         "content": "Do you ever have those moments where somebody is so mind shatteringly selfish and horrible that it may take you a few days to even  put the right words together, because it is so unfathomable for most of the population to be that way?? Yet they think they are logical and normal?? Even though everybody around them has to put effort into tolerating being around them??? No matter what damage they cause in their wake they will NEVER change anything or say they are sorry???",
         "type": "Generic"
      }
   ],
   "title": "Thread Title",
   "is_still_participant": true,
   "thread_type": "Regular",
   "thread_path": "OtherPerson_ea77ad6afb"
}
