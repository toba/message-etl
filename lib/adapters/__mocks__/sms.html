<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>First Last-Hyphen</title>
<style type="text/css">
          /* Copyright 2011 Google Inc.  All Rights Reserved. */

body {
  font-size: 13px;
  font-family: Arial, Helvectica, sans-serif;
}

a {
  color: #00c;
}

a:hover {
  text-decoration: underline;
}

.tags,
.hChatLog,
.noteContainer {
  margin: 0 auto;
  width: 750px;
  min-width: 750px;
}

.message {
  max-width: 640px;
}

/* The following three style rules exist to override default behavior in the
 * user-agent stylesheet used by common browsers such as Chrome. */
cite {
  font-style: normal;
}

q::before {
  content: "";
}

q::after {
  content: "";
}

.tags,
.noteContainer {
  margin-top: 13px;
}

.participants {
  margin-bottom: 13px;
}

        </style></head>
<body><div class="hChatLog hfeed">
<div class="message"><abbr class="dt" title="2013-06-17T20:46:00.861-04:00">Jun 17, 2013, 8:46:00 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Flight 2244, 15 minutes delayed, also had to check in bag because Carolyn bought Kayla and I stuff from bath &amp; body works</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T20:48:20.247-04:00">Jun 17, 2013, 8:48:20 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>Have good flying</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T22:47:28.109-04:00">Jun 17, 2013, 10:47:28 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Landed</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T22:47:58.180-04:00">Jun 17, 2013, 10:47:58 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>We&#39;ll drive up the hill and await your text</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T22:48:46.831-04:00">Jun 17, 2013, 10:48:46 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Ok</q>
</div> <div class="message"><abbr class="dt" title="2013-06-17T23:00:17.481-04:00">Jun 17, 2013, 11:00:17 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>Outside</q>
</div> <div class="message"><abbr class="dt" title="2013-06-22T15:08:16.526-04:00">Jun 22, 2013, 3:08:16 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>At Bonneville point</q>
</div> <div class="message"><abbr class="dt" title="2013-06-22T15:09:13.122-04:00">Jun 22, 2013, 3:09:13 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12087242867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>So far so good, nice day for a ride</q>
</div> <div class="message"><abbr class="dt" title="2013-06-26T19:51:05.573-04:00">Jun 26, 2013, 7:51:05 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+12085508050"><abbr class="fn" title="">Me</abbr></a></cite>:
<q>Superman? Home in 15</q>
</div></div>

<div class="tags">Labels:
<a rel="tag" href="http://www.google.com/voice#inbox">Inbox</a>, <a rel="tag" href="http://www.google.com/voice#sms">Text</a></div></body></html>