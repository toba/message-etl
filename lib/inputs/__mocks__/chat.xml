<?xml version="1.0"?>
<?xml-stylesheet type='text/xsl' href='MessageLog.xsl'?>
<Log LogonName="other@trailimage.com" FirstSessionID="1" LastSessionID="26">
   <Message Date="6/24/2003" Time="10:24:17 AM" DateTime="2003-06-24T16:24:17.396Z" SessionID="1">
      <From>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </From>
      <To>
         <User LogonName="self@trailimage.com" FriendlyName="Self Person"/>
      </To>
      <Text Style="font-family:Microsoft Sans Serif; font-weight:bold; font-style:italic; color:#008080; ">let me guess, working hard again?</Text>
   </Message>
   <Message Date="6/24/2003" Time="10:24:42 AM" DateTime="2003-06-24T16:24:42.102Z" SessionID="1">
      <From>
         <User LogonName="self@trailimage.com" FriendlyName="Self Person"/>
      </From>
      <To>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </To>
      <Text Style="font-family:Tahoma; color:#800080; ">yes</Text>
   </Message>
   <Message Date="6/24/2003" Time="10:25:54 AM" DateTime="2003-06-24T16:25:54.216Z" SessionID="1">
      <From>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </From>
      <To>
         <User LogonName="sunflower77h@hotmail.com" FriendlyName="butterfly"/>
         <User LogonName="self@trailimage.com" FriendlyName="Jason Abbott"/>
         <User LogonName="momofhailey@hotmail.com" FriendlyName="don't you wish you were floating the tubes!"/>
      </To>
      <Text Style="font-family:Microsoft Sans Serif; font-weight:bold; font-style:italic; color:#008080; ">Laura wearing you out yet?</Text>
   </Message>
   <Message Date="6/24/2003" Time="10:29:06 AM" DateTime="2003-06-24T16:29:06.953Z" SessionID="1">
      <From>
         <User LogonName="self@trailimage.com" FriendlyName="Self Person"/>
      </From>
      <To>
         <User LogonName="other@trailimage.com" FriendlyName="Other Person"/>
      </To>
      <Text Style="font-family:Tahoma; color:#800080; ">sorry, been on and still on the phone</Text>
   </Message>
</Log>