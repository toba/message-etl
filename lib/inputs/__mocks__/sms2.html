<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>First Last-Hyphen</title>
<style type="text/css">
          /* Copyright 2011 Google Inc.  All Rights Reserved. */

body {
  font-size: 13px;
  font-family: Arial, Helvectica, sans-serif;
}

a {
  color: #00c;
}

a:hover {
  text-decoration: underline;
}

.tags,
.hChatLog,
.noteContainer {
  margin: 0 auto;
  width: 750px;
  min-width: 750px;
}

.message {
  max-width: 640px;
}

/* The following three style rules exist to override default behavior in the
 * user-agent stylesheet used by common browsers such as Chrome. */
cite {
  font-style: normal;
}

q::before {
  content: "";
}

q::after {
  content: "";
}

.tags,
.noteContainer {
  margin-top: 13px;
}

.participants {
  margin-bottom: 13px;
}

        </style></head>
<body><div class="hChatLog hfeed">
<div class="message"><abbr class="dt" title="2012-09-25T19:19:11.272-04:00">Sep 25, 2012, 7:19:11 PM
Eastern Time</abbr>:
<cite class="sender vcard"><a class="tel" href="tel:+867"><span class="fn">First Last-Hyphen</span></a></cite>:
<q>At park for a bit, signed papers</q>
</div></div>

<div class="tags">Labels:
<a rel="tag" href="http://www.google.com/voice#inbox">Inbox</a>, <a rel="tag" href="http://www.google.com/voice#sms">Text</a></div></body></html>