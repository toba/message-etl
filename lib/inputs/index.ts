export { facebookMessenger as fb } from './messenger';
export { googleVoice as gv } from './google-voice';
export { mailbox as mb } from './mbox';
export { microsoftChat as chat } from './chat';
export { passport } from './passport';
