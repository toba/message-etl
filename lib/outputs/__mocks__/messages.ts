import { lipsum } from '@toba/test';
import { Message, Relation, Source } from '../../types';

export const messages: Message[] = [
   {
      from: Relation.Self,
      on: new Date(2014, 5, 12),
      text: lipsum,
      source: Source.FacebookMessenger
   }
];
