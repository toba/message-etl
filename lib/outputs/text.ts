import { Message, Writer } from '../types';

export const serialize = (m: Message): string => `${m.on.getHours()}:${
   m.on.getMinutes
}
   ${m.from}
   ${m.text}`;

export const text: Writer = {
   write(messages: Message[]) {
      

      const list: string = messages.map(serialize).join('\n');
      return `<!DOCTYPE html><html><body>${list}</body></html>`;
   },
   serialize,
   fileName: 'messages.txt'
};
