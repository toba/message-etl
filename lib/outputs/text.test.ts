import '@toba/test';
import { readFileText } from '@toba/test';

import { fb } from '../inputs/';
import { text } from './text';
import { messages } from './__mocks__/messages';

jest.mock('../config/index');

test('serializes message', () => {
   expect(text.serialize(messages[0])).toMatchSnapshot();
});

test('generates delimited file', async () => {
   const raw = await readFileText(
      `${__dirname}/../inputs/__mocks__/message.json`
   );
   const messages = fb.process(raw);
   const file = text.write(messages);

   expect(file).toMatchSnapshot();
});
