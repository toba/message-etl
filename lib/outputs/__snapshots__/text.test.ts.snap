// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates delimited file 1`] = `
"<!DOCTYPE html><html><body><article>
   <div>2</div>
   <time>Tue Oct 16 2018 12:12:53 GMT-0600 (Mountain Daylight Time)</time>
   <p>Oh, I know what that means. Ok</p>
</article><article>
   <div>1</div>
   <time>Tue Oct 16 2018 12:06:50 GMT-0600 (Mountain Daylight Time)</time>
   <p>Maybe I translated wrong. Full message: \\"Its 3160 Lewis River Rd Woodland Wa 98674 for know it will be changing when the new houses and road goes in don't when that will happen.\\"</p>
</article><article>
   <div>2</div>
   <time>Tue Oct 16 2018 12:06:27 GMT-0600 (Mountain Daylight Time)</time>
   <p>What do you mean not in the new place?</p>
</article><article>
   <div>2</div>
   <time>Tue Oct 16 2018 12:05:57 GMT-0600 (Mountain Daylight Time)</time>
   <p>Jessica: 󰀀</p>
</article><article>
   <div>1</div>
   <time>Sun Oct 14 2018 15:46:30 GMT-0600 (Mountain Daylight Time)</time>
   <p>You sent a video.</p>
</article><article>
   <div>1</div>
   <time>Tue Oct 16 2018 11:53:18 GMT-0600 (Mountain Daylight Time)</time>
   <p>And he spelled \\"your\\" wrong</p>
</article><article>
   <div>1</div>
   <time>Sat Oct 13 2018 15:12:46 GMT-0600 (Mountain Daylight Time)</time>
   <p>You: 󰀀</p>
</article><article>
   <div>1</div>
   <time>Tue Oct 16 2018 11:53:02 GMT-0600 (Mountain Daylight Time)</time>
   <p>Scott says the address is still 3160 Lewis River Rd. Not in the new place yet.</p>
</article><article>
   <div>2</div>
   <time>Tue Oct 16 2018 09:51:34 GMT-0600 (Mountain Daylight Time)</time>
   <p>Rough</p>
</article><article>
   <div>1</div>
   <time>Tue Oct 16 2018 09:45:52 GMT-0600 (Mountain Daylight Time)</time>
   <p>Went to make latte for first time ever here and we're out of milk. What a dump.</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 20:41:34 GMT-0600 (Mountain Daylight Time)</time>
   <p>Jessica: 󰀀</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 20:41:16 GMT-0600 (Mountain Daylight Time)</time>
   <p>Was on perch a while</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 20:41:08 GMT-0600 (Mountain Daylight Time)</time>
   <p>Mellow</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 20:41:05 GMT-0600 (Mountain Daylight Time)</time>
   <p>Seems to be</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 20:40:39 GMT-0600 (Mountain Daylight Time)</time>
   <p>She feeling okay?</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 20:39:53 GMT-0600 (Mountain Daylight Time)</time>
   <p>Cats new hiding spot</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 20:39:45 GMT-0600 (Mountain Daylight Time)</time>
   <p>You sent a photo.</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 19:24:44 GMT-0600 (Mountain Daylight Time)</time>
   <p>So very sad</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 19:22:48 GMT-0600 (Mountain Daylight Time)</time>
   <p>Felt so sad going in his room to turn off the light, looking around</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 18:31:42 GMT-0600 (Mountain Daylight Time)</time>
   <p>You: 󰀀</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 18:31:33 GMT-0600 (Mountain Daylight Time)</time>
   <p>Oh, I know it was 3160. I'll find out</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 18:30:32 GMT-0600 (Mountain Daylight Time)</time>
   <p>Meant as more of a question :)</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 18:29:29 GMT-0600 (Mountain Daylight Time)</time>
   <p>Thanks</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 18:27:50 GMT-0600 (Mountain Daylight Time)</time>
   <p>3170 Lewis River Rd
Woodland, WA 98674</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 18:27:49 GMT-0600 (Mountain Daylight Time)</time>
   <p>The address I have is</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 16:48:19 GMT-0600 (Mountain Daylight Time)</time>
   <p>You: 󰀀</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 16:47:12 GMT-0600 (Mountain Daylight Time)</time>
   <p>I'll be by to get Hunter around 5:30</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 16:46:15 GMT-0600 (Mountain Daylight Time)</time>
   <p>Waiting in line for cat</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 13:56:54 GMT-0600 (Mountain Daylight Time)</time>
   <p>yeah ...</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 13:56:35 GMT-0600 (Mountain Daylight Time)</time>
   <p>no clue on the xbox, sold is probably a good bet. He can deal with Scott on that one</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 13:54:26 GMT-0600 (Mountain Daylight Time)</time>
   <p>didn't see xbox anywhere. i guess he sold it? or I put it somewhere I totally forgot? but i don't think so. just had it in the office, locked.</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 13:53:40 GMT-0600 (Mountain Daylight Time)</time>
   <p>checked his room. it seemed clean in the sense that there aren't clothes on the floor anymore.</p>
</article><article>
   <div>2</div>
   <time>Mon Oct 15 2018 13:53:14 GMT-0600 (Mountain Daylight Time)</time>
   <p>If he's back by 4:30 and his room is clean and he's totally packed</p>
</article><article>
   <div>1</div>
   <time>Mon Oct 15 2018 13:42:55 GMT-0600 (Mountain Daylight Time)</time>
   <p>Hunter wanting to go visit friends. I said to check with you.</p>
</article><article>
   <div>1</div>
   <time>Sat Jun 22 2013 12:03:31 GMT-0600 (Mountain Daylight Time)</time>
   <p>This is my route
https://dl.dropboxusercontent.com/u/33199511/lava-mountain-loop.jpg</p>
</article><article>
   <div>2</div>
   <time>Mon Jun 10 2013 22:27:23 GMT-0600 (Mountain Daylight Time)</time>
   <p>Good night, sweet dreams reality is obviously not your forte.</p>
</article><article>
   <div>2</div>
   <time>Mon Jun 10 2013 08:03:18 GMT-0600 (Mountain Daylight Time)</time>
   <p>SPOILED monster.</p>
</article><article>
   <div>2</div>
   <time>Sun Jun 09 2013 09:55:02 GMT-0600 (Mountain Daylight Time)</time>
   <p>That is hilarious, we are all laughing. Hitting the road now</p>
</article><article>
   <div>2</div>
   <time>Wed Feb 23 2011 21:47:01 GMT-0700 (Mountain Standard Time)</time>
   <p>Do you ever have those moments where somebody is so mind shatteringly selfish and horrible that it may take you a few days to even  put the right words together, because it is so unfathomable for most of the population to be that way?? Yet they think they are logical and normal?? Even though everybody around them has to put effort into tolerating being around them??? No matter what damage they cause in their wake they will NEVER change anything or say they are sorry???</p>
</article></body></html>"
`;

exports[`serializes message 1`] = `
"<article>
   <div>1</div>
   <time>Thu Jun 12 2014 00:00:00 GMT-0600 (Mountain Daylight Time)</time>
   <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
</article>"
`;
